handler/handlers.go

```
@@ -125,6 +127,15 @@ func (b *BookHandler) toBook(book *openlibrarydto.Book) (bm *model.Book) {
 	if book.Author != nil {
 		author = book.Author[0].Name
 	}
+	yearRgx := regexp.MustCompile(`[0-9]{4}`)
+
+	yearRgx.MatchString(book.Year)
+	yearString := yearRgx.FindString(book.Year)
+
+	year, err := strconv.Atoi(yearString)
+	if err != nil {
+		fmt.Println("error while converting year from string to int", err)
```

You want to return the error, not just print it.